lib/rpmsg/rpmsg_virtio.c

```diff
@@ -411,7 +411,7 @@ static void rpmsg_virtio_rx_callback(struct virtqueue *vq)
 			status = ept->cb(ept, RPMSG_LOCATE_DATA(rp_hdr),
 					 rp_hdr->len, rp_hdr->src, ept->priv);
 
-			RPMSG_ASSERT(status == RPMSG_SUCCESS,
+			RPMSG_ASSERT(status >= RPMSG_SUCCESS,
```

Comment:
Could you explain why the status will be > RPMSG_SUCCESS? Thanks.

Reply:
For example, at the end of many cb implemenation, it normally send the response to another side by rpmsg_send, but rpmsg_send return the data length for success, negative error code for fail. Without this change, I have to:
int xxx_cb(...)
{
  int ret = rpmsg_send(...);
  return ret >= 0 ? 0 : ret;
}
Instead the more clean code:
int xxx_cb(...)
{
  return rpmsg_send(...);
}

Reply:
I see, just the rpmsg_send() in cb is more like application specific implementation. The cb() should return what's the cb expect to return.
From your change it is strange to see it still considers the return as successful even the return value is not RPMSG_SUCCESS. I think the rpmsg.h should add comments to explain what's the expected result.

Reply:
Ok, the new patch add comment just before rpmsg_ept_cb and change RPMSG_SUCCESS to 0.